File: backend/src/main/java/com/back/ourlog/domain/genre/entity/DiaryGenre.java
package com.back.ourlog.domain.genre.entity;

import com.back.ourlog.domain.diary.entity.Diary;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
@IdClass(DiaryGenreId.class)
public class DiaryGenre {
    @Id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "diary_id", nullable = false)
    private Diary diary;

    @Id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "genre_id", nullable = false)
    private Genre genre;

    public DiaryGenre(Diary diary, Genre genre) {
        this.diary = diary;
        this.genre = genre;
    }
}

File: backend/src/main/java/com/back/ourlog/domain/genre/entity/DiaryGenre.kt
package com.back.ourlog.domain.genre.entity

import com.back.ourlog.domain.diary.entity.Diary
import jakarta.persistence.*

@Entity
@IdClass(DiaryGenreId::class)
class DiaryGenre(

    @Id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "diary_id", nullable = false)
    val diary: Diary,

    @Id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "genre_id", nullable = false)
    val genre: Genre
)
